ntp-udp/src/lib.rs

```diff
@@ -296,14 +388,162 @@ fn recv(
     Ok((bytes_read, sock_addr, None))
 }
 
+fn fetch_send_timestamp_help(
+    socket: &std::net::UdpSocket,
+    expected_counter: u32,
+) -> io::Result<Option<NtpTimestamp>> {
+    // we get back two control messages: one with the timestamp (just like a receive timestamp),
+    // and one error message with no error reason. The payload for this second message is kind of
+    // undocumented.
+    //
+    // section 2.1.1 of https://www.kernel.org/doc/Documentation/networking/timestamping.txt says that
+    // a `sock_extended_err` is returned, but in practice we also see a socket address. The linux
+    // kernel also has this https://github.com/torvalds/linux/blob/master/tools/testing/selftests/net/so_txtime.c#L153=
+    const CONTROL_SIZE: usize = control_message_space::<[libc::timespec; 3]>()
+        + control_message_space::<(libc::sock_extended_err, libc::sockaddr)>();
```

Comment:
as a note: `sockaddr` gives a tight bound for current test cases. `sockaddr_storage` is much bigger, but I'm not sure that is ever really needed (sockaddr should cover ipv6 already)